### ISSUE TITLE
podman rm external-container: sometimes needs -f

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#8735]
There seem to be two types of buildah containers: completed-successfully ones, and incomplete ones. `podman ps -a --storage` can see both, but can only `rm` the latter if `-f` is given.

Complete containers:
```console
$ buildah from quay.io/libpod/testimage:20200929
testimage-working-container
$ ./bin/podman rm testimage-working-container
testimage-working-container
$ echo $?
0     <----- yay it works
```

Incomplete containers, created by interrupting a build:
```console
$ printf "FROM quay.io/libpod/testimage:20200929\nRUN sleep 30\n"| timeout -v 5 podman build -t thiswillneverexist -
STEP 1: FROM quay.io/libpod/testimage:20200929
STEP 2: RUN sleep 30
timeout: sending signal TERM to command ‘podman’
$ echo $?
124        <--- expected

$ ./bin/podman rm testimage-working-container
Error: no container with name or ID testimage-working-container found: no such container
$ ./bin/podman ps -a --storage
CONTAINER ID  IMAGE                              COMMAND  CREATED         STATUS   PORTS   NAMES
57a7769d2ca2  quay.io/libpod/testimage:20200929  buildah  56 seconds ago  storage          testimage-working-container
$ ./bin/podman rm 57a7769d2ca2
Error: no container with name or ID 57a7769d2ca2 found: no such container
$ echo $?
1

$ ./bin/podman rm -f 57a7769d2ca2        <--- add "-f", it works
57a7769d2ca2
$ echo $?
0
```
master @ 9379ee97907266e5a96bd9c4ae178e0e572a6f2a

Terminology: "complete" and "incomplete" is my simpleminded attempt to find a way to describe this situation. My nomenclature may be incorrect or even misleading. Please go by the reproducer, not my words.

**Comment by mheon** [comment]
Guess: the ones that won't `rm` without `-f` are mounted - I'd suspect a check of `ps` would show some extra fuse-overlay processes corresponding to them.

I think that not removing mounted containers without `-f` is probably correct - it doesn't catch all partially-failed containers from a `podman build` but it does make it a bit more difficult to accidentally remove a CRI-O container someone is actually using.

**Comment by edsantiago** [comment]
Good guess:
```
esm       989027  0.0  0.0   5080  1948 ?        Ss   05:05   0:00 /usr/bin/fuse-overlayfs -o lowerdir=/home/esm/.local/share/containers/storage/overlay/l/SW7WJA3PCSQ7U7SZLWWPEL4SE5,upperdir=/home/esm/.local/share/containers/storage/overlay/eb4005632e8a6a9ecf51901025c00bd655226ebafdebe3c8424e29951355ddb1/diff,workdir=/home/esm/.local/share/containers/storage/overlay/eb4005632e8a6a9ecf51901025c00bd655226ebafdebe3c8424e29951355ddb1/work,context="system_u:object_r:container_file_t:s0:c184,c434" /home/esm/.local/share/containers/storage/overlay/eb4005632e8a6a9ecf51901025c00bd655226ebafdebe3c8424e29951355ddb1/merged
```
It's an orphan, PPID is 1. I can't umount it, probably due to namespace issues. Killing the process has no effect on podman's `rm` behavior.

If this is the cause, I wonder if podman could issue a better diagnostic than `no such container`? 

Thank you @mheon. I do appreciate your insights, but I also sincerely hope that you are spending your PTO time in a personally enjoyable manner.

**Comment by rhatdan** [comment]
It looks like we failed to remove the external container and then fail over to attempting to remove the "podman" containers, which tells us the container does not exists.

**Comment by rhatdan** [comment]
@edsantiago what do you think a good outcome for this issue would be?

**Comment by edsantiago** [comment]
I still think this is weird and unexpected behavior. Some possible fixes I can think of:
* Make `podman rm xxx` handle it, without `-f`
* Add new `podman rm --storage xxx` option. This at least makes some sense given that `podman ps -a --storage` is what is necessary to show the container
* Improve the error message, e.g. `Error: storage containers can only be removed using --force option` ("no such container" error is just nonsensical from the perspective of a user who has just seen that container in `ps` output)

**Comment by rhatdan** [comment]
Would this satisfy your issue?

./bin/podman rm testimage-working-container-1
Error: container "testimage-working-container-1" is mounted and cannot be removed without using force: container state improper

**Comment by edsantiago** [comment]
Moderate preference for explicitly including the dashes in `--force`, but yes, I'd be happy with that

**Comment by edsantiago** [comment]
Closed by #9014
